aptos-move/block-executor/src/scheduler.rs

```diff
@@ -272,7 +281,7 @@ pub struct Scheduler {
     /// Shared marker that is set when a thread detects that all txns can be committed.
     done_marker: CachePadded<AtomicBool>,
 
-    has_halted: CachePadded<AtomicBool>,
+    halted_status: CachePadded<Mutex<HaltedStatus>>,
```

Comment:
will this have performance issues? 

alternativelly, you could have an enum with incremental values for each, that you store in AtomicU8?

Reply:
Only called at most once by each worker, that's why I didn't optimize too much.